Monika After Story/game/script-topics.rpy

```diff
@@ -12905,3 +12905,205 @@ label monika_enjoyingspring:
     else:
         m 2rkc "...but I guess there's no real way to avoid it, is there?"
     return
+
+# start of bookmark and derandom related events
+default persistent._mas_player_bookmarked = {}
+# dict to store bookmarked events
+default persistent._mas_player_derandomed = {}
+# dict to store player derandomed events
+default persistent.flagged_monikatopic = None
+# var set when we flag a topic for derandom
+
+init python:
+    def derandom_topic():
+        """
+        Function for the derandom hotkey, 'x'
+        """
+        curr_topic = persistent.current_monikatopic
+        if (
+            mas_getEV(curr_topic) is not None
+            and mas_getEV(curr_topic).random
+            and curr_topic.startswith("monika_")
+            and mas_getEV(curr_topic).prompt != mas_getEV(curr_topic).eventlabel
```

Comment:
what kinds of events does this fall under, where the prompt == event label? I can imagine maybe the calender ones, but should at least add a comment here saying what events this is excluding.

also evaluate if this conditional check should be its own function (since you do it in the bookmark topic as well), as that could get an appropriate name and make it clearer as to what this is excluding

Reply:
it's checking to make sure it's an event that actually has a prompt, since if no prompt is given, it defaults to the label. just trying to ensure nothing without a prompt is included.

Reply:
do we have any `monika_` topics that werent given a prompt? In general that prefix is supposed to reserved for topics with prompts, so this might be an unecessary check